integration/service/test/geopm_report_cron_plot.py

```diff
@@ -0,0 +1,242 @@
+        style={
+            'textAlign': 'center',
+            'color': self._colors['page_text'],
+            'font-family': self._font_family
+        }
+        app.layout = html.Div(style={'backgroundColor': self._colors['background']}, children=[
+            html.H1(children='GEOPM Power Report', style=style),
+            dcc.Graph(id='gpu-power',
+                      figure=self.plot_power('gpu', begin_date, end_date)),
+            dcc.Graph(id='cpu-power',
+                      figure=self.plot_power('cpu', begin_date, end_date)),
+            dcc.Graph(id='dram-power',
+                      figure=self.plot_power('dram', begin_date, end_date)),
+            dcc.Graph(id='energy-pie',
+                      figure=self.plot_energy(begin_date, end_date)),
+        ])
+        app.run(host='0.0.0.0', port=port)
+
+    def static_webpage(self, begin_date, end_date, output_html):
+        if (not output_html.endswith('.html')):
+            raise RuntimeError(f'Output file {output_html} does not end in .html')
+        header = f'<h1 style="text-align: center; color: {self._colors["page_text"]}; background-color: {self._colors["background"]}; '\
+                 f'font-family: {self._font_family};">GEOPM Power Report</h1>'
+        with open(output_html, 'w') as fid:
+            fid.write('<html>\n    <body>\n')
+            fid.write(header)
+            for domain in ('gpu', 'cpu', 'dram'):
+                fig = self.plot_power(domain, begin_date, end_date)
+                fid.write(fig.to_html(full_html=False, default_height='450px', include_plotlyjs='cdn'))
+            fig = self.plot_energy(begin_date, end_date)
+            fid.write(fig.to_html(full_html=False, default_height='450px', include_plotlyjs='cdn'))
+            fid.write('    </body>\n</html>\n')
+
+def get_parser():
+    parser = ArgumentParser(description=main.__doc__)
+    parser.add_argument('report', nargs='*')
+    static_group = parser.add_mutually_exclusive_group()
+    static_group.add_argument('-s', '--static-html', dest='static_html', default=None,
+                              help='Do not run web server, output static html into directory provided')
+    static_group.add_argument('-p', '--port', dest='port', type=int, default=10850,
+                              help='Port opened by server to provide visualization')
+    return parser
+
+def main():
+    """Command line interface for creating geopmsession report visualizations
+
+    Parses a collection of geopmsession YAML reports and creates an HTML
+    visualiation.  By default it will open a dash server on port 10850.  Can
+    also create static web pages with the ``--static-html`` option.
+
+    """
+    err = 0
+    try:
+        args = get_parser().parse_args()
+        if len(args.report) == 0:
+            args.print_help()
+            return -1
+        if len(args.report) == 1:
+            files = args.report[0]
+        else:
+            files = args.report
```

Comment:
A singleton list is conditionally flattened here, but later code performs type checks to see whether we need to iterate a list of strings or operate on one string. Why not just keep this as a list so the other code does not need to perform those checks?

The current implementation accepts either a single glob spec or 2+ paths. Alternatively it could accept a list of globs.